auto.def

```diff
@@ -106,6 +106,8 @@ options {
   with-zlib:path            => "Location of zlib"
   zstd=0                    => "Enable Zstandard header cache compression support"
   with-zstd:path            => "Location of Zstandard"
+# MacOS
+  osxkeychain=0             => "Use MacOS's Keychain for authentication"
```

Comment:
I think we should use `macos` instead of `osx`, both here and in the file names.

Reply:
Agreed. Will make the change.